easybuild/framework/easyblock.py

```diff
@@ -3510,6 +3515,7 @@ def make_module_step(self, fake=False):
             txt += self.make_module_extend_modpath()
             txt += self.make_module_req()
             txt += self.make_module_extra()
+            txt += ''.join(set(exts_module_extra))
```

Comment:
We already have a mechanism to let extensions add stuff to the module file, via `self.module_extra_extensions`, see https://github.com/easybuilders/easybuild-framework/blob/14ed92242498105d8c61cabab1bf32889256c0d1/easybuild/framework/easyblock.py#L1785-L1788, which is then picked up in `make_module_extra_extensions` at https://github.com/easybuilders/easybuild-framework/blob/14ed92242498105d8c61cabab1bf32889256c0d1/easybuild/framework/easyblock.py#L1334-L1335.

So all that's needed in the `reticulate` easyblock is to make the `run` method return the extra lines that should be included in the module file.

Important detail here: `self.module_extra_extensions` is currently not updated yet in `install_extensions_parallel` (only in `install_extensions_sequential`), which means the extra stuff for the module file will not be picked up when using `--experimental --parallel-extensions-install`.

That should definitely be fixed if we start relying on it for `R` packages... (right now, only the `numpy` easyblock does a `return` in the `run` method to inject stuff into the module file, and `--parallel-extensions-install` doesn't work yet for Python packages, so it wasn't a big issue up until now).

Reply:
that's clear, thanks!